File: 0383-ransom-note/0383-ransom-note.py
class Solution(object):
    def canConstruct(self, ransomNote, magazine):
        """
        :type ransomNote: str
        :type magazine: str
        :rtype: bool
        """
        ransomKeys = {}
        magazineKeys ={}
        for a in ransomNote:
            ransomKeys[a] = ransomKeys.get(a,0)+1
        for a in magazine:
            magazineKeys[a] = magazineKeys.get(a,0)+1

        print(ransomKeys,magazineKeys )
        for key in ransomKeys:
            if magazineKeys.get(key) < ransomKeys.get(key):
                return False

        return True

File: 0383-ransom-note/0383-ransom-note.scala
import scala.collection.mutable


object Solution {

    def canConstruct(ransomNote: String, magazine: String): Boolean = {

        var rhm = mutable.HashMap[Char, Int]()
        var mhm = mutable.HashMap[Char, Int]()

        for(char<-ransomNote){
            rhm.put(char, rhm.getOrElse(char, 0)+1)
        }

        for(char<-magazine){
        mhm.put(char, mhm.getOrElse(char,0)+1)
        }

        var output=true
        for((k,v)<-rhm if output){
        if(mhm.getOrElse(k,0)<v){
            output=false
        }
        }
    
        output
  }
}